### PR TITLE
Various AdHoc constants value fixes

### PR DESCRIPTION
**Pull request by jxchinaren**
AdHoc query fix:
1. Treat constants in SQL literal numbers either as integer or decimal.
2. Add tests for float E sign.
3. Fix the various of AdHoc planner caching with type issues.
4. Refactor and add more testing utils.
---
Missing stored procedure fix (Can be done separately from this ticket):
5. Stored procedure constants literal value fix either integer or decimal.
6. More tests on out of range of our decimal literal value if needed.

### CODE REVIEW

File: src/frontend/org/voltdb/planner/ParameterizationInfo.java [hunk 1]
```diff
@@ -177,16 +177,31 @@ static void parameterizeRecursively(VoltXMLElement node,
...
+                VoltType vt = VoltType.typeFromString(typeStr);
+
                 String value = null;
-                if (VoltType.typeFromString(typeStr) != VoltType.NULL) {
+                if (vt != VoltType.NULL) {
                     value = node.attributes.get("value");
                 }
                 paramValues.add(value);
+
+                if (vt == VoltType.NUMERIC) {
```
Comment (BillWhite) [inline src/frontend/org/voltdb/planner/ParameterizationInfo.java]: I think you are trying to see if this numeric value is an integer or a decimal.  If that is right,
perhaps you should add a comment that says so.  What if the string is a float?  What if the
string is not properly formatted, say it's "1.3.4"?

**Comment by BillWhite** [comment]
Done with this comment round.

File: src/frontend/org/voltdb/planner/ParameterizationInfo.java [hunk 2]
```diff
@@ -177,16 +177,31 @@ static void parameterizeRecursively(VoltXMLElement node,
...
+                    vt = VoltType.BIGINT;
+                    try {
+                        Long.parseLong(value);
+                    } catch (NumberFormatException e) {
+                        // DECIMAL probably is not bigger/smaller enough than our decimal range.
```
Comment (jxchinaren) [inline src/frontend/org/voltdb/planner/ParameterizationInfo.java]: Should change the comment here also:
"Our DECIMAL may not be bigger/smaller enough to store the constant value"
